arcgis-ios-sdk-samples/Features/Display feature layers/DisplayFeatureLayersViewController.swift

```diff
@@ -0,0 +1,183 @@
+        geodatabase = AGSGeodatabase(name: "LA_Trails")
+        
+        // Load the geodatabase for feature tables.
+        geodatabase.load { [weak self] (error: Error?) in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                // Get the feature table with the name.
+                let featureTable = self.geodatabase.geodatabaseFeatureTable(withName: "Trailheads")!
+                // Create a feature layer with the feature table.
+                let featureLayer = AGSFeatureLayer(featureTable: featureTable)
+                // Set the viewpoint to Malibu, California.
+                let viewpoint = AGSViewpoint(latitude: 34.0772, longitude: -118.7989, scale: 6e5)
+                self.setMap(featureLayer: featureLayer, viewpoint: viewpoint)
+            }
+        }
+    }
+    
+    /// Load a feature layer with a local geopackage.
+    func loadGeopackage() {
+        // Create a geopackage from a named bundle resource.
+        geoPackage = AGSGeoPackage(name: "AuroraCO")
+        
+        // Load the geopackage.
+        geoPackage.load { [weak self] (error: Error?) in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                // Add the first feature layer from the geopackage to the map.
+                if let featureTable = self.geoPackage?.geoPackageFeatureTables.first {
+                    // Create the feature layer with the feature table.
+                    let featureLayer = AGSFeatureLayer(featureTable: featureTable)
+                    // Set the viewpoint to Aurora, Colorado.
+                    let viewpoint = AGSViewpoint(latitude: 39.7294, longitude: -104.8319, scale: 5e5)
+                    self.setMap(featureLayer: featureLayer, viewpoint: viewpoint)
+                }
+            }
+        }
+    }
+    
+    /// Load a feature layer with a local shapefile.
+    func loadShapefile() {
+        // Create a shapefile feature table from a named bundle resource.
+        let shapefileTable = AGSShapefileFeatureTable(name: "ScottishWildlifeTrust_ReserveBoundaries_20201102")
+        
+        // Create a feature layer for the shapefile feature table.
+        let featureLayer = AGSFeatureLayer(featureTable: shapefileTable)
+        // Set the viewpoint to Scotland.
+        let viewpoint = AGSViewpoint(latitude: 56.641344, longitude: -3.889066, scale: 6e6)
+        setMap(featureLayer: featureLayer, viewpoint: viewpoint)
+    }
+    
+    /// Add the feature layer to the map and set the viewpoint.
+    /// - Parameters:
+    ///   - featureLayer: The `AGSFeatureLayer` to display and add to the map.
+    ///   - viewpoint: The `AGSViewpoint` to change the map to.
+    func setMap(featureLayer: AGSFeatureLayer, viewpoint: AGSViewpoint) {
```

Comment:
Methinks the name `setMap` is a bit confusing because it doesn't change the actual map and it also sets the viewpoint, which isn't a property of the map. I would suggest renaming it to `setFeatureLayer(_:viewpoint:)`.